packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -148,18 +163,28 @@ class AnalyzeCommand extends FlutterCommand {
 
     if (argResults['current-package']) {
       // ./lib/main.dart
-      String mainPath = 'lib/main.dart';
+      String mainPath = path.join('lib', 'main.dart');
       if (FileSystemEntity.isFileSync(mainPath)) {
         dartFiles.add(mainPath);
         pubSpecDirectories.add('.');
         foundAnyInCurrentDirectory = true;
       }
+
+      // Check for <package>/lib/<package>.dart.
+      String packageName = path.basename(Directory.current.path);
+      String mainFilePath = path.join('lib', '$packageName.dart');
```

Comment:
Ah, well I'm assuming that if it contains a `lib` dir then the dir is likely a pub package, and the dir name == the package name. I'm banking on conventions here.

We could also pull the package name out of the pubspec file, if one exists.